packages/web3-utils/src/big_decimal.ts

```diff
@@ -0,0 +1,51 @@
+export class BigDecimal {
+	private readonly bigint: bigint;
+	private readonly decimals: number;
+
+	public constructor(value: bigint | string, decimals?: number) {
+		if (typeof value === 'string' && decimals) {
+			throw new Error('Can not specify decimal with string value');
+		}
+
+		if (typeof value === 'string') {
+			try {
+				const [integer, fraction] = String(value).split('.').concat('');
```

Comment:
What is the purpose of `.concat('')`?

```typescript
'4.20'.split('.') // > (2) ['4', '20']
'4.20'.split('.').concat('') // (3) ['4', '20', '']
```

Reply:
The purpose was to avoid have a `null` value for `fraction`. A comment is going to be added to clarify intention